python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -160,6 +160,56 @@ interface {{ ethernet_interface.name }}
+{%                 if ethernet_interface.encapsulation_vlan.client.inner_protocol is arista.avd.defined and ethernet_interface.encapsulation_vlan.client.inner_protocol != client_protocol %}
+{%                     set encapsulation_cli = 'client ' ~ client_protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.client.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.client.inner_protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.client.inner %}
+{%                 else %}
+{%                     set encapsulation_cli = 'client ' ~ client_protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.client.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.client.inner %}
+{%                 endif %}
+{%                 if ethernet_interface.encapsulation_vlan.network.protocol | arista.avd.default == 'client inner' %}
+{%                     set network_flag = True %}
+{%                     set encapsulation_cli = encapsulation_cli ~ ' network ' ~ ethernet_interface.encapsulation_vlan.network.protocol %}
+{%                 endif %}
+{%             endif %}
+{%         elif client_protocol in ['untagged', 'unmatched'] %}
+{%             set encapsulation_cli = 'client ' ~ ethernet_interface.encapsulation_vlan.client.protocol %}
+{%         endif %}
+{%         if encapsulation_cli is arista.avd.defined %}
+{%             if client_protocol in ['dot1q', 'dot1ad', 'untagged'] and ethernet_interface.encapsulation_vlan.network.protocol is arista.avd.defined and not network_flag %}
+{%                 set network_protocol = ethernet_interface.encapsulation_vlan.network.protocol %}
+{%                 if ethernet_interface.encapsulation_vlan.network.vlan is arista.avd.defined %}
+{%                     set encapsulation_cli = encapsulation_cli ~ ' network ' ~ network_protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.network.vlan %}
+{%                 elif ethernet_interface.encapsulation_vlan.network.inner is arista.avd.defined and ethernet_interface.encapsulation_vlan.network.outer is arista.avd.defined %}
+{%                     if ethernet_interface.encapsulation_vlan.network.inner_protocol is arista.avd.defined and ethernet_interface.encapsulation_vlan.network.inner_protocol != network_protocol %}
+{%                         set encapsulation_cli = encapsulation_cli ~ ' network ' ~ network_protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.network.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.network.inner_protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.network.inner %}
+{%                     else %}
+{%                         set encapsulation_cli = encapsulation_cli ~ ' network ' ~ network_protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.network.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.network.inner %}
```

Comment:
```suggestion
{%                     set encapsulation_cli = encapsulation_cli ~ ' network ' ~ network_protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.network.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.network.inner_protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.network.inner %}
{%                 endif %}
```

Reply:
I think eos shortcuts should be avoided. @MaheshGSLAB could you confirm.

Reply:
Here, if the `protocol` and `inner-protocol` are same it doesnot write the `inner_protocol` in config and if they are different it configures it .
```
s1-leaf2(config-if-Et2.1-encap-vlan)#client dot1q outer 10 inner dot1q 25 
s1-leaf2(config-if-Et2.1-encap-vlan)#show active
interface Ethernet2.1
   encapsulation vlan
      client dot1q outer 10 inner 25
s1-leaf2(config-if-Et2.1-encap-vlan)#client dot1q outer 10 inner dot1ad 25
s1-leaf2(config-if-Et2.1-encap-vlan)#show active
interface Ethernet2.1
   encapsulation vlan
      client dot1q outer 10 inner dot1ad 25
```

Reply:
If the CLI is valid (meaning EOS accepts it) then we should not build any "fancy logic" in here. As long as it is possible to generate config matching the running config of all cases, then we are good.

Reply:
Fixed